### ISSUE TITLE
gvaclassify do not work on model license-plate-recognition-barrier-0001

### ISSUE THREAD

**Issue by IvanZhangDoIt** [openvinotoolkit/dlstreamer_gst#213]
Hi, I am investigating the capability of DL Streamer to create our IVA solution and met some basic issues using the intel model.  Could you help me out on this? Thanks

**[Issue Description]**

- Test license-plate recognition model with the following command:
- All examples work well in: /opt/intel/openvino_2021.3.436/data_processing/dl_streamer/samples/gst_launch

`
gst-launch-1.0 filesrc location=/opt/intel/openvino_2021.3.436/deployment_tools/demo/car_1.bmp ! decodebin ! video/x-raw ! videoconvert ! gvadetect model=/root/intel/dl_streamer/models/intel/vehicle-license-plate-detection-barrier-0106/FP32/vehicle-license-plate-detection-barrier-0106.xml model-proc=./model_proc/vehicle-license-plate-detection-barrier-0106.json device=CPU ! queue ! gvaclassify model=/root/intel/dl_streamer/models/intel/license-plate-recognition-barrier-0001/FP32/license-plate-recognition-barrier-0001.xml model-proc=./model_proc/license-plate-recognition-barrier-0001.json device=CPU object-class=license-plate ! queue ! gvametaconvert json-indent=4 ! gvametapublish method=file file-path=log_101.json file-format=json ! fakesink sync=false
`

gvadetection works well, but gvaclassify not work. 

- the output file log_101.json is :

`
[{
    "objects": [
        {
            "detection": {
                "bounding_box": {
                    "x_max": 0.6801894307136536,
                    "x_min": 0.3101099133491516,
                    "y_max": 0.7327951192855835,
                    "y_min": 0.18795567750930786
                },
                "confidence": 1.0,
                "label": "vehicle",
                "label_id": 1
            },
            "h": 347,
            "roi_type": "vehicle",
            "w": 277,
            "x": 232,
            "y": 120
        },
        {
            "detection": {
                "bounding_box": {
                    "x_max": 0.5247715711593628,
                    "x_min": 0.4413079023361206,
                    "y_max": 0.6849339008331299,
                    "y_min": 0.6443006992340088
                },
                "confidence": 0.9428476691246033,
                "label": "license-plate",
                "label_id": 2
            },
            "h": 26,
            "roi_type": "license-plate",
            "w": 63,
            "x": 331,
            "y": 410
        }
    ],
    "resolution": {
        "height": 637,
        "width": 749
    }
}]
` 
- log with setting : GST_DEBUG=2,gvaclassify:7
`
0:00:00.030517543  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:135:gst_gva_classify_init:<GstGvaClassify@0x562e591e24c0> gst_gva_classify_init
0:00:00.030546743  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:136:gst_gva_classify_init:<GstGvaClassify@0x562e591e24c0> (null)
0:00:00.030551843  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:156:gst_gva_classify_cleanup:<GstGvaClassify@0x562e591e24c0> gva_classify_cleanup
Setting pipeline to PAUSED ...
0:00:00.043975894  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
0:00:00.044742903  2804 0x562e591e5160 WARN                 basesrc gstbasesrc.c:3600:gst_base_src_start_complete:<filesrc0> pad not activated yet
Pipeline is PREROLLING ...
0:00:00.063649916  2804 0x562e591f2180 WARN                 default inference_impl.cpp:420:InferenceImpl:<gvadetect0> Loading model: device=CPU, path=/root/intel/dl_streamer/models/intel/vehicle-license-plate-detection-barrier-0106/FP32/vehicle-license-plate-detection-barrier-0106.xml
0:00:00.063672016  2804 0x562e591f2180 WARN                 default inference_impl.cpp:422:InferenceImpl:<gvadetect0> Initial settings batch_size=1, nireq=0
0:00:00.064217222  2804 0x562e591f2180 WARN              GVA_common model_proc_parser.cpp:55:parseProcessingItem: "layer_name" field has not been set. Its value will be defined as ANY
0:00:00.253784155  2804 0x562e591f20c0 WARN                 default inference_impl.cpp:420:InferenceImpl:<gvaclassify0> Loading model: device=CPU, path=/root/intel/dl_streamer/models/intel/license-plate-recognition-barrier-0001/FP32/license-plate-recognition-barrier-0001.xml
0:00:00.253818355  2804 0x562e591f20c0 WARN                 default inference_impl.cpp:422:InferenceImpl:<gvaclassify0> Initial settings batch_size=1, nireq=0
0:00:00.309204978  2804 0x562e591f20c0 DEBUG            gvaclassify gstgvaclassify.c:213:on_base_inference_initialized:<gvaclassify0> on_base_inference_initialized
0:00:00.317207969  2804 0x562e591f2060 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
Pipeline is PREROLLED ...
Setting pipeline to PLAYING ...
0:00:00.318657585  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
New clock: GstSystemClock
Got EOS from element "pipeline0".
Execution ended after 0:00:00.000284303
Setting pipeline to PAUSED ...
0:00:00.319815198  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
Setting pipeline to READY ...
0:00:00.321276614  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
Setting pipeline to NULL ...
Freeing pipeline ...
0:00:00.321901821  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:172:gst_gva_classify_finalize:<gvaclassify0> finalize
0:00:00.321911521  2804 0x562e591e5160 DEBUG            gvaclassify gstgvaclassify.c:156:gst_gva_classify_cleanup:<gvaclassify0> gva_classify_cleanup

`

- The following method has tried:

1) change input from .bmp to .png file or .mp4
2) change gvaclassify to gvainference  and set inference-region=1
3) add gvatrack and gvaclassify for car attribution. 
'
gst-launch-1.0 filesrc location=/opt/intel/openvino_2021.3.436/deployment_tools/demo/car_1.bmp ! decodebin ! videoconvert ! gvadetect model=/root/intel/dl_streamer/models/intel/vehicle-license-plate-detection-barrier-0106/FP32/vehicle-license-plate-detection-barrier-0106.xml model-proc=./model_proc/vehicle-license-plate-detection-barrier-0106.json threshold=0.6 device=CPU ! queue ! gvatrack tracking-type=short-term ! queue ! gvaclassify model=/root/intel/dl_streamer/models/intel/vehicle-attributes-recognition-barrier-0039/FP32/vehicle-attributes-recognition-barrier-0039.xml model-proc=./model_proc/vehicle-attributes-recognition-barrier-0039.json reclassify-interval=10 device=CPU object-class=vehicle ! queue ! gvaclassify model=/root/intel/dl_streamer/models/intel/license-plate-recognition-barrier-0001/FP32/license-plate-recognition-barrier-0001.xml model-proc=./model_proc/license-plate-recognition-barrier-0001.json device=CPU object-class=license-plate ! queue ! gvametaconvert json-indent=4 ! gvametapublish method=file file-path=log_20:30:01.json file-format=json ! fakesink sync=false
'
result:
'
[{
    "objects": [
        {
            "color": {
                "label": "black",
                "model": {
                    "name": "vehicle-attributes-recognition-barrier-0039"
                }
            },
            "detection": {
                "bounding_box": {
                    "x_max": 0.6801894307136536,
                    "x_min": 0.3101099133491516,
                    "y_max": 0.7327951192855835,
                    "y_min": 0.18795567750930786
                },
                "confidence": 1.0,
                "label": "vehicle",
                "label_id": 1
            },
            "h": 347,
            "id": 1,
            "roi_type": "vehicle",
            "type": {
                "label": "car",
                "model": {
                    "name": "vehicle-attributes-recognition-barrier-0039"
                }
            },
            "w": 277,
            "x": 232,
            "y": 120
        },
        {
            "detection": {
                "bounding_box": {
                    "x_max": 0.5247715711593628,
                    "x_min": 0.4413079023361206,
                    "y_max": 0.6849339008331299,
                    "y_min": 0.6443006992340088
                },
                "confidence": 0.9428476691246033,
                "label": "license-plate",
                "label_id": 2
            },
            "h": 26,
            "id": 2,
            "roi_type": "license-plate",
            "w": 63,
            "x": 331,
            "y": 410
        }
    ],
    "resolution": {
        "height": 637,
        "width": 749
    }
}]

'
log
'
0:00:00.036013705  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:135:gst_gva_classify_init:<GstGvaClassify@0x560f0a60a4c0> gst_gva_classify_init
0:00:00.036046806  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:136:gst_gva_classify_init:<GstGvaClassify@0x560f0a60a4c0> (null)
0:00:00.036052406  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:156:gst_gva_classify_cleanup:<GstGvaClassify@0x560f0a60a4c0> gva_classify_cleanup
0:00:00.036069606  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:53:gst_gva_classify_set_property:<gvaclassify0> set_property
0:00:00.036120707  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:135:gst_gva_classify_init:<GstGvaClassify@0x560f0a60a8b0> gst_gva_classify_init
0:00:00.036126607  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:136:gst_gva_classify_init:<GstGvaClassify@0x560f0a60a8b0> (null)
0:00:00.036131207  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:156:gst_gva_classify_cleanup:<GstGvaClassify@0x560f0a60a8b0> gva_classify_cleanup
Setting pipeline to PAUSED ...
0:00:00.055167821  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify1> get_property
0:00:00.055270622  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
0:00:00.056667638  2839 0x560f0a632b80 WARN                 basesrc gstbasesrc.c:3600:gst_base_src_start_complete:<filesrc0> pad not activated yet
Pipeline is PREROLLING ...
0:00:00.076342959  2839 0x560f0a624060 WARN                 default inference_impl.cpp:420:InferenceImpl:<gvadetect0> Loading model: device=CPU, path=/root/intel/dl_streamer/models/intel/vehicle-license-plate-detection-barrier-0106/FP32/vehicle-license-plate-detection-barrier-0106.xml
0:00:00.076365559  2839 0x560f0a624060 WARN                 default inference_impl.cpp:422:InferenceImpl:<gvadetect0> Initial settings batch_size=1, nireq=0
0:00:00.076981166  2839 0x560f0a624060 WARN              GVA_common model_proc_parser.cpp:55:parseProcessingItem: "layer_name" field has not been set. Its value will be defined as ANY
0:00:00.265935091  2839 0x560f0a624180 WARN                 default inference_impl.cpp:420:InferenceImpl:<gvaclassify0> Loading model: device=CPU, path=/root/intel/dl_streamer/models/intel/vehicle-attributes-recognition-barrier-0039/FP32/vehicle-attributes-recognition-barrier-0039.xml
0:00:00.265962292  2839 0x560f0a624180 WARN                 default inference_impl.cpp:422:InferenceImpl:<gvaclassify0> Initial settings batch_size=1, nireq=0
0:00:00.300286378  2839 0x560f0a624180 DEBUG            gvaclassify gstgvaclassify.c:213:on_base_inference_initialized:<gvaclassify0> on_base_inference_initialized
0:00:00.300525680  2839 0x560f0a624120 WARN                 default inference_impl.cpp:420:InferenceImpl:<gvaclassify1> Loading model: device=CPU, path=/root/intel/dl_streamer/models/intel/license-plate-recognition-barrier-0001/FP32/license-plate-recognition-barrier-0001.xml
0:00:00.300558381  2839 0x560f0a624120 WARN                 default inference_impl.cpp:422:InferenceImpl:<gvaclassify1> Initial settings batch_size=1, nireq=0
0:00:00.353323274  2839 0x560f0a624120 DEBUG            gvaclassify gstgvaclassify.c:213:on_base_inference_initialized:<gvaclassify1> on_base_inference_initialized
0:00:00.360496455  2839 0x560f0a6240c0 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify1> get_property
0:00:00.360665657  2839 0x560f0a6240c0 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
Pipeline is PREROLLED ...
Setting pipeline to PLAYING ...
0:00:00.362379976  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify1> get_property
0:00:00.362563378  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
New clock: GstSystemClock
Got EOS from element "pipeline0".
Execution ended after 0:00:00.000354304
Setting pipeline to PAUSED ...
0:00:00.363972994  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify1> get_property
0:00:00.364090395  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
Setting pipeline to READY ...
0:00:00.366118618  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify1> get_property
0:00:00.366272220  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:84:gst_gva_classify_get_property:<gvaclassify0> get_property
Setting pipeline to NULL ...
Freeing pipeline ...
0:00:00.367084129  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:172:gst_gva_classify_finalize:<gvaclassify1> finalize
0:00:00.367097429  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:156:gst_gva_classify_cleanup:<gvaclassify1> gva_classify_cleanup
0:00:00.369615757  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:172:gst_gva_classify_finalize:<gvaclassify0> finalize
0:00:00.369634558  2839 0x560f0a632b80 DEBUG            gvaclassify gstgvaclassify.c:156:gst_gva_classify_cleanup:<gvaclassify0> gva_classify_cleanup

'

**[Environment]**

- Docker image:
docker.io/openvino/ubuntu20_data_dev         2021.3_vaapi_fix 
- OpenVino version:
openvino_2021.3.436

**Comment by adranit** [comment]
Hi @IvanZhangDoIt 
Could you set parameter "add-tensor-data=true" for gvametaconvert?
gvametaconvert json-indent=4 add-tensor-data=true
This parameter add raw tensors data to json file.

I suspect that the model can't recognize number from license-plate. Do you use images with Chinese license-plate?

Thanks,
Anton

**Comment by mliu2020** [comment]
I hope @adranit's tip could solve your problem.

If not, could you clarify the behavior when you said "gvaclassify not work", can you see any characters? Also want to double check, this is the same model and json file from [security_barrier_camera_demo](https://github.com/openvinotoolkit/open_model_zoo/tree/master/demos/security_barrier_camera_demo/cpp), they were working with the demo successfully.

Mark

**Comment by IvanZhangDoIt** [comment]
> Hi @IvanZhangDoIt
> Could you set parameter "add-tensor-data=true" for gvametaconvert?
> gvametaconvert json-indent=4 add-tensor-data=true
> This parameter add raw tensors data to json file.
> 
> I suspect that the model can't recognize number from license-plate. Do you use images with Chinese license-plate?
> 
> Thanks,
> Anton

Thanks, @adranit. I add "add-tensor-data = True" and get the following result which contain the incorrect result with all data is -1.0.
And I do use the image "car_1.bmp" which is the example in the installation package for   demo "demo_security_barrier_camera" and I run the C++ demo, it works very well. The result is just strange when I used gvaclassify to load the model and do the inference. 

`
[{
    "objects": [
        {
            "color": {
                "label": "black",
                "model": {
                    "name": "vehicle-attributes-recognition-barrier-0039"
                }
            },
            "detection": {
                "bounding_box": {
                    "x_max": 0.6801894307136536,
                    "x_min": 0.3101099133491516,
                    "y_max": 0.7327951192855835,
                    "y_min": 0.18795567750930786
                },
                "confidence": 1.0,
                "label": "vehicle",
                "label_id": 1
            },
            "h": 347,
            "id": 1,
            "roi_type": "vehicle",
            "tensors": [
                {
                    "confidence": 1.0,
                    "label_id": 1,
                    "layer_name": "DetectionOutput_",
                    "layout": "ANY",
                    "model_name": "vehicle-license-plate-detection-barrier-0106",
                    "name": "detection",
                    "precision": "UNSPECIFIED"
                },
                {
                    "layout": "ANY",
                    "name": "object_id",
                    "precision": "UNSPECIFIED"
                },
                {
                    "confidence": 0.9999995231628418,
                    "data": [
                        1.6703570748788366e-09,
                        4.376173379228021e-08,
                        1.0972116548657596e-09,
                        3.9026130593811104e-07,
                        2.7261448654058995e-09,
                        1.2223162926261466e-08,
                        0.9999995231628418
                    ],
                    "label": "black",
                    "label_id": 6,
                    "layer_name": "color",
                    "layout": "NCHW",
                    "model_name": "vehicle-attributes-recognition-barrier-0039",
                    "name": "color",
                    "precision": "FP32"
                },
                {
                    "confidence": 1.0,
                    "data": [
                        1.0,
                        4.585236270532311e-13,
                        8.194281841608642e-13,
                        1.6043050568570472e-11
                    ],
                    "label": "car",
                    "label_id": 0,
                    "layer_name": "type",
                    "layout": "NCHW",
                    "model_name": "vehicle-attributes-recognition-barrier-0039",
                    "name": "type",
                    "precision": "FP32"
                }
            ],
            "type": {
                "label": "car",
                "model": {
                    "name": "vehicle-attributes-recognition-barrier-0039"
                }
            },
            "w": 277,
            "x": 232,
            "y": 120
        },
        {
            "detection": {
                "bounding_box": {
                    "x_max": 0.5247715711593628,
                    "x_min": 0.4413079023361206,
                    "y_max": 0.6849339008331299,
                    "y_min": 0.6443006992340088
                },
                "confidence": 0.9428476691246033,
                "label": "license-plate",
                "label_id": 2
            },
            "h": 26,
            "id": 2,
            "roi_type": "license-plate",
            "tensors": [
                {
                    "confidence": 0.9428476691246033,
                    "label_id": 2,
                    "layer_name": "DetectionOutput_",
                    "layout": "ANY",
                    "model_name": "vehicle-license-plate-detection-barrier-0106",
                    "name": "detection",
                    "precision": "UNSPECIFIED"
                },
                {
                    "layout": "ANY",
                    "name": "object_id",
                    "precision": "UNSPECIFIED"
                },
                {
                    "data": [
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0
                    ],
                    "layer_name": "decode",
                    "layout": "NCHW",
                    "model_name": "LPRNet",
                    "name": "license-plate",
                    "precision": "FP32"
                }
            ],
            "w": 63,
            "x": 331,
            "y": 410
        }
    ],
    "resolution": {
        "height": 637,
        "width": 749
    }
}]

`

**Comment by IvanZhangDoIt** [comment]
> I hope @adranit's tip could solve your problem.
> 
> If not, could you clarify the behavior when you said "gvaclassify not work", can you see any characters? Also want to double check, this is the same model and json file from [security_barrier_camera_demo](https://github.com/openvinotoolkit/open_model_zoo/tree/master/demos/security_barrier_camera_demo/cpp), they were working with the demo successfully.
> 
> Mark

@mliu2020  Thanks Mark. I have built the demo you mentioned get the correct license plate.  Then I think it should work with the same model and image with DL Streamer pipeline.  The difference is that I load the model by "gvaclassify" and using the post-proc  in  **license-plate-recognition-barrier-0001.json**, while the demo you mentioned load the with Inference Engine c++ API. 



`
        "converter": "tensor_to_label",
        "method": "index"
`

**Comment by IvanZhangDoIt** [comment]
And If I change the sink part to 

```shell
"gvawatermark ! videoconvert ! fpsdisplaysink video-sink=ximagesink sync=false"
```

I can see the bbox of license plate is correct, it just can not be recognized by the model "license-plate-recognition-barrier-0001.xml", and I load the pre/post procs json (license-plate-recognition-barrier-0001.json) from Open Model Zoo:

**Comment by mliu2020** [comment]
@IvanZhangDoIt,

Are you saying to replace "fakesink" with "gvawatermark ! videoconvert ! fpsdisplaysink video-sink=ximagesink sync=false"?

This time, the pipeline works, but it has the correct bbox and doesn't show the characters.

In your previous case when you mentioned "gvaclassify not work", was it the same behavior? Can you give me more detail about the error? It would be better if you can have detailed descriptions.

Mark

**Comment by IvanZhangDoIt** [comment]
> @IvanZhangDoIt,
> 
> Are you saying to replace "fakesink" with "gvawatermark ! videoconvert ! fpsdisplaysink video-sink=ximagesink sync=false"?
> 
> This time, the pipeline works, but it has the correct bbox and doesn't show the characters.
> 
> In your previous case when you mentioned "gvaclassify not work", was it the same behavior? Can you give me more detail about the error? It would be better if you can have detailed descriptions.
> 
> Mark

@mliu2020 

Sink part means the GStreamer pipeline sink part (Source->Filters->Sink), I just want to emphasize car and plate detection works well but plate recognition(gvaclassify), same result. The result was verified if I output the result to json file with:

```shell
gvametaconvert json-indent=4 ! gvametapublish method=file file-path=log_101.json file-format=json ! fakesink sync=false
```
or display(of course, I create mp4 video with the car_1.bmp) 

```shell
"gvawatermark ! videoconvert ! fpsdisplaysink video-sink=ximagesink sync=false"
```

**Comment by IvanZhangDoIt** [comment]
@mliu2020 @adranit Thanks.
I can give more detail why I think my pipeline should work and why I say gvaclassify do not work with the model

1.  What did I do with gst-launch-1.0 pipeline?
The gst-launch-1.0 pipeline: 

```
 ->gvadetect with  model: vehicle-license-plate-detection-barrier-0106.xml   works
 ->gvaclassify with model: vehicle-attributes-recognition-barrier-0039.xml     works
 ->gvaclassify with model: license-plate-recognition-barrier-0001.xml             can not get expected result.
```

2. Why should I think the pipeline should work? This is because the example in the following which has the same pipeline and use same models.
https://docs.openvinotoolkit.org/2018_R5/_samples_security_barrier_camera_demo_README.html
The difference is that my pipeline is based on DL Streamer, the example just used Inference Engine API directly.
I use the same image with the demp - car_1.bmp.

3. Are these models supported in DL Streamer? Yes, it provide the json pre/post proc file.
https://github.com/openvinotoolkit/dlstreamer_gst/wiki/supported-models

- Have you guys tried on your side? Few of my colleagues tried it, I don't think it relates to environment.
- Is the json proc file correctly? I do not see any instruction on how to write or test the json proc file, what I can do is to read the gvaclassify source code and samples to try to understand. 
- Did I miss something like configuration for in the gst-launch? 
- I have read the source code of the security barrier demo, it did some tricky things after detection before plate recognition. Maybe I need try by adding a gvapython elements before gvaclassify?
```c++
        for (Detector::Result result : results) {
            switch (result.label) {
                case 1:
                {
                    vehicleRects.emplace_back(result.location & cv::Rect{cv::Point(0, 0), sharedVideoFrame->frame.size()});
                    break;
                }
                case 2:
                {
                    // expanding a bounding box a bit, better for the license plate recognition
                    result.location.x -= 5;
                    result.location.y -= 5;
                    result.location.width += 10;
                    result.location.height += 10;
                    plateRects.emplace_back(result.location & cv::Rect{cv::Point(0, 0), sharedVideoFrame->frame.size()});
                    break;
                }
                default: throw std::exception();  // must never happen
                         break;
            }
        }
```

**Comment by IvanZhangDoIt** [comment]
I created a plate flie here.
![plate_1](https://user-images.githubusercontent.com/25627631/123012365-39431a00-d416-11eb-9c2d-6edcd3a821d3.png)
and create a very simple pipeline to test the gvaclassify with it. 

```shell
gst-launch-1.0 filesrc location=/tmp/plate_1.png ! decodebin ! videoconvert ! queue ! gvaclassify model=/root/intel/dl_streamer/models/intel/license-plate-recognition-barrier-0001/FP32/license-plate-recognition-barrier-0001.xml model-proc=./model_proc/license-plate-recognition-barrier-0001.json inference-region=0 device=CPU ! queue ! gvametaconvert json-indent=4 add-tensor-data=true ! gvametapublish method=file file-path=log_983036736.json file-format=json ! fakesink sync=false
```

get the result

```json
[{
    "objects": [
        {
            "h": 31,
            "tensors": [
                {
                    "data": [
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0,
                        -1.0
                    ],
                    "layer_name": "decode",
                    "layout": "NCHW",
                    "model_name": "LPRNet",
                    "name": "license-plate",
                    "precision": "FP32"
                }
            ],
            "w": 69,
            "x": 0,
            "y": 0
        }
    ],
    "resolution": {
        "height": 31,
        "width": 69
    },
    "timestamp": 0
}]

```

**Comment by mliu2020** [comment]
Hi @IvanZhangDoIt,

Could you do a quick check the same pipeline in OV2021.2?

It seems I can do the recognition on this release.

Mark

**Comment by IvanZhangDoIt** [comment]
Thanks Mark @mliu2020 

It works on "openvino_2021.2.185"  with docker image:  docker.io/openvino/ubuntu18_data_dev:2021.2   
but NOT in "openvino_2021.3.436"  with docker image: docker.io/openvino/ubuntu20_data_dev:2021.3_vaapi_fi

**Comment by mliu2020** [comment]
Thanks so much for the confirm,

We will check this clue in our side for the root cause.

Mark

**Comment by adranit** [comment]
@IvanZhangDoIt ,
The root cause of the issue is wrong initial values of an auxiliary blob *seq_ind* that required by the model for decoding. Requirements for initial values were changed in 2021.3 and we did not notice. Fix of the issue is included in #216.
Thanks for using DL Streamer!

https://docs.openvinotoolkit.org/2021.3/omz_models_model_license_plate_recognition_barrier_0001.html
https://docs.openvinotoolkit.org/2021.2/omz_models_model_license_plate_recognition_barrier_0001.html

**Comment by IvanZhangDoIt** [comment]
@adranit 

Thanks

**Comment by adranit** [comment]
fixed in 2021.4.1
